File: for-local-test/python/409.longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count = [0] * 128
        for char in s:
            count[ord(char)] += 1
        result = 0
        hasOdd = False
        for c in count:
            if c % 2 == 0:
                result += c
            else:
                result += c - 1
                hasOdd = True
        return result * 2 + (1 if hasOdd else 0)
    
if __name__ == "__main__":
    solution = Solution()
    input_str = "abccccdd"
    longest_palindrome_length = solution.longestPalindrome(input_str)
    print("Longest Palindrome Length:", longest_palindrome_length)

File: for-local-test/409.longest-palindrome.java

class Solution {
    public int longestPalindrome(String s) {
        int[] count = new int[128];
        for (char c : s.toCharArray()) {
            count[c]++;
        }
        int result = 0;
        boolean hasOdd = false;
        for (int c : count){
            if (c % 2 == 0) {
                result += c / 2;
            } else{
                result += c / 2;
                hasOdd = true;
            }
        }
        return result * 2 + (hasOdd ? 1 : 0);
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        String input = "abccccdd";
        int longestPalindromeLength = solution.longestPalindrome(input);
        System.out.println("Longest Palindrome Length: " + longestPalindromeLength);
    }
}


